### ISSUE TITLE
Cannot call a method on a null-valued expression

### ISSUE THREAD

**Issue by shahmohamod** [clinthuffman/PAL#32]
Overall progress... Status: Creating charts, PAL 2.8.2 Progress: 76.4705882352941%... Creating charts, PhysicalDisk Write Latency Analysis

An error occurred on...
For ($i=0;$i -le $iValues.GetUpperBound(0);$i++)
At C:\Program Files\PAL\PAL\PAL.ps1:3041 char:15 + For ($i=0;$i -le $iValues.GetUpperBound(0);$i++) + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
You cannot call a method on a null-valued expression.

SCRIPT ARGUMENTS:
Log: C:\PerfLogs\20181207\DCX\SPW-MDY-020\18H00-19H30\PAL_System_Overview.blg
ThresholdFile: C:\Program Files\PAL\PAL\SystemOverview.xml
AnalysisInterval: AUTO
IsOutputHtml: $True
IsOutputXml: $False
HtmlOutputFileName: [LogFileName]_PAL_ANALYSIS_[DateTimeStamp].htm
XmlOutputFileName: [LogFileName]_PAL_ANALYSIS_[DateTimeStamp].xml
OutputDir: C:\PerfLogs\20181207\DCX\SPW-MDY-020\18H00-19H30\PAL Reports
AllCounterStats: $False
NumberOfThreads: 4
IsLowPriority: $True
DisplayReport: True

**Comment by clinthuffman** [comment]
Can you send me the log at %temp%\PAL.log or upload your counter log to http://bit.ly/PalUpload?

**Comment by Veronica-A1** [comment]
Hi,
I get the same error but when a colleague of mine logs in on the same server, running the exact same PAL Wizard (PAL_FlatFiles_2.8.1) with the exact same settings (blg, Auto-Detect Threshold File etc) there is no error. 

We are confused. What is missing on my windows account? Does anyone have any idea? blg file is from C-drive so no network involved.

Br Veronica

**Comment by jeffstokes72** [comment]
is the file on root of c: or in a directory? Are you both members of
Administrators, same account privileges, etc?

**Comment by clinthuffman** [comment]
Take a look at %temp%\PAL.log and see what lead up to the error or upload your counter log to http://bit.ly/PalUpload and I can debug it. Common problems are if special characters are in the path. And as Jeff mentioned if the counter log is on the root of C: drive, then there may be problems. Try moving the counter log (*.blg) to your Desktop folder and running PAL.

**Comment by Veronica-A1** [comment]
> is the file on root of c: or in a directory? Are you both members of
> Administrators, same account privileges, etc?
> …
Thank you for your quick response. The blg file was not in c root, it was placed on desktop, first desktop for my user and then copied to desktop for my colleague when he was logged in. As far as I know we are both included in the same administrator group. What I don't know is if my colleague is also included in another group with yet more permissions. Do you have any hint what to look for? Particular work folder for PAL wizard? It is the very same blg file at the same location (but not really the same since it is desktop for each user).

**Comment by Veronica-A1** [comment]
> Take a look at %temp%\PAL.log and see what lead up to the error or upload your counter log to http://bit.ly/PalUpload and I can debug it. Common problems are if special characters are in the path. And as Jeff mentioned if the counter log is on the root of C: drive, then there may be problems. Try moving the counter log (*.blg) to your Desktop folder and running PAL.

If you don't have any problem using PAL wizard you probably won't run into my error I'm afraid. Can special characters differ when we are logged in as different users running the very same blg file in the very same PAL wizard?

I try to find %temp%\PAL.log but in that path I have no such file.

**Comment by clinthuffman** [comment]
Yes, things like dashes (-) and periods (.) can potentially cause problems with the Powershell interpreter. %temp% is an environment variable that will go to C:\Users\<user>\AppData\Local\Temp folder. When PAL is run, it creates at PAL.log file there.

**Comment by Veronica-A1** [comment]
I understand %temp% is extracted. When I look into that folder for each one of us I see:
{GUID}_RunPAL.bat
PalAutoDetectThresholdFile_GUID.xml
PalAutoDetectObjectsFromCounterLogFile_GUID.txt
A folder having a GUID as folder name containing:
_FilteredPerfmonLog.csv
CounterListFilter.txt

I searched for pal.log in c drive and d drive but no such file.

I am not allowed to upload my blg files bacuase of server names. I have run at least 14 different blg files from different servers with the same error all the time. When my colleague logs in with his windows account it works fine for all of them with no error.

I put the blg file in D:\PAL\test\file.blg and tried again, no file copy and no user specific folder. Still the same problem for me, not for him.

**Comment by RusalkaLesna** [comment]
> Hi,
> I get the same error but when a colleague of mine logs in on the same server, running the exact same PAL Wizard (PAL_FlatFiles_2.8.1) with the exact same settings (blg, Auto-Detect Threshold File etc) there is no error.
> 
> We are confused. What is missing on my windows account? Does anyone have any idea? blg file is from C-drive so no network involved.
> 
> Br Veronica

I had exact same issue. In Windows settings I had to change Regional format to English (US)

**Comment by Veronica-A1** [comment]
RusalkaLesna that was it! I hade Swedish in regional settings. It was English windows so I didn't think about that. Thanks a lot! And thank you all for your help with ideas!

**Comment by clinthuffman** [comment]
Ah, sorry about that. I try to globalize PAL, but I've only tested it in English. Thank you, RusalkaLesna.
